layer/trace_layer.cpp

```diff
@@ -51,7 +52,16 @@ const VkLayerProperties kLayerProps = {
         GFXRECON_PROJECT_VERSION_DESIGNATION
 };
 
-const std::vector<VkExtensionProperties> kDeviceExtensionProps = { VkExtensionProperties{ "VK_EXT_tooling_info", 1 } };
+const std::vector<VkExtensionProperties> kDeviceExtensionProps = {
+    VkExtensionProperties{ "VK_EXT_tooling_info", 1 },
+    VkExtensionProperties{ VK_EXT_DEBUG_MARKER_EXTENSION_NAME, VK_EXT_DEBUG_MARKER_SPEC_VERSION }
+};
+
+std::unordered_set<std::string> kProvidedDeviceFunctions = { "vkCmdDebugMarkerBeginEXT",
+                                                             "vkCmdDebugMarkerEndEXT",
+                                                             "vkCmdDebugMarkerInsertEXT",
+                                                             "vkDebugMarkerSetObjectNameEXT",
+                                                             "vkDebugMarkerSetObjectTagEXT" };
 
```

Comment:
`find(...) != ...end()` on a `set` is idiomatic and seems fine to me.  This array could get longer, so we'd have to worry about where the crossover point is, and you'd have to verify clang and MSVC also behave the same way.  While I commend your thoroughness including writing a test case in QuickBench (I learned that exists today!), we have a lot of work to do and I don't think this change important enough to microbenchmark something that occurs once at application startup.

I agree it could be `const`.

My only thought about this new variable would be that in the functions below, `kProvidedDeviceFunctions.count(in) > 0` is slightly more readable to me until `unordered_set::contains` in C++20.